org.wso2.codequalitymatrices/src/main/java/com/wso2/code/quality/matrices/CodeQualityMatricesExecutor.java

```diff
@@ -0,0 +1,106 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.wso2.code.quality.matrices;
+
+import com.google.gson.Gson;
+import org.apache.log4j.Logger;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static com.wso2.code.quality.matrices.model.Constants.COMMITS_INSIDE_GIVEN_PATCH;
+
+/**
+ * This is used for creating executing the program.
+ *
+ * @since 1.0.0
+ */
+public class CodeQualityMatricesExecutor {
+    private static final Logger logger = Logger.getLogger(CodeQualityMatricesExecutor.class);
+
+    private final String pmtToken;
+    private final String patchId;
+    private final String gitHubToken;
+
+    /**
+     * This create an instance of CodeQualityMatricesExecutor class.
+     *
+     * @param pmtToken    PMT Access Token
+     * @param patchId     Patch ID
+     * @param gitHubToken Github access token
+     */
+    public CodeQualityMatricesExecutor(String pmtToken, String patchId, String gitHubToken) {
+        this.pmtToken = pmtToken;
+        this.patchId = patchId;
+        this.gitHubToken = gitHubToken;
+    }
+
+    /**
+     * This is the entry point to this application.
+     */
+    public void execute() {
+        try {
+            List<String> commitHashes = findCommitHashesInPatch();
+            ChangesFinder changesFinder = new ChangesFinder();
+            Set<String> authorCommits = changesFinder.obtainRepoNamesForCommitHashes(gitHubToken, commitHashes);
+            ReviewAnalyser reviewAnalyser = new ReviewAnalyser();
+            reviewAnalyser.findReviewers(authorCommits, gitHubToken);
+            reviewAnalyser.printReviewUsers();
+            logger.debug("The application executed successfully");
+        } catch (CodeQualityMatricesException e) {
+            logger.debug(e.getMessage(), e.getCause());
+        }
+    }
+
+    /**
+     * This is used to filter out the commit hashes that belongs to the given patch
+     *
+     * @return List of commithashes contained in the given patch
+     */
+    private List<String> findCommitHashesInPatch() throws CodeQualityMatricesException {
+        PmtApiCaller pmtApiCaller = new PmtApiCaller();
+        String jsonText;
+        List<String> commitHashes = new ArrayList<>();
+        try {
+            jsonText = pmtApiCaller.callApi(pmtToken, patchId);
+        } catch (CodeQualityMatricesException e) {
+            throw new CodeQualityMatricesException("Error occurred while calling PMT API", e);
+        }
+        Gson gson = new Gson();
+        if (jsonText != null) {
```

Comment:
can we move gson initialization into if(jsonText != null){} block? so if the condition failed no need to create such a instance as it is used only inside that IF block.